CleverRaven/Cataclysm-DDA
# Seasonal Sound Effects

#### Summary
None

#### Purpose of change
Fixes #52295.

Added the ability to specify seasons for sound effects, so that different sounds can play depending on the current season.

#### Describe the solution
This solution is a bit different than what was suggested in the issue. It implements a new optional field `"season"` for sound effects, where the possible values are:
- "spring"
- "summer"
- "autumn"
- "winter"

This will flag the sound to only play during the specified season.

Example:
```json
{
  "type": "sound_effect",
  "id": "smash_success",
  "variant": "t_gates_mech_control_lab",
  "season": "winter",
  "volume": 50,
  "files": [
    "terrain/metal/smash_metal_1.ogg",
    "terrain/metal/smash_metal_2.ogg",
    "terrain/metal/smash_metal_3.ogg",
    "terrain/metal/smash_metal_4.ogg",
    "terrain/metal/smash_metal_5.ogg",
    "terrain/metal/smash_metal_6.ogg",
    "terrain/metal/smash_metal_7.ogg",
    "terrain/metal/smash_metal_8.ogg",
    "terrain/metal/smash_metal_9.ogg",
    "terrain/metal/smash_metal_10.ogg"
  ]
}
```

#### Describe alternatives you've considered
The issue suggests using an id extension, similar to tileset variants. I wasn't sure how integrating the season into the id would affect the sound metadata, so I opted for the above solution instead. This can be changed if the issue's suggested solution is more appropriate.

#### Testing
1. Copy an existing soundpack into data/sound/ ([Otopack](https://github.com/Kenan2000/Otopack-Mods-Updates) in this case)
2. Ensure that sounds effects still play without any "season" tags (i.e.: opening a door)
3. Add `"season": "spring"` to open_door sounds
4. Open a door in-game during Spring, sound still plays
5. Change "spring" to "summer" on open_door sounds
6. Open a door in-game during Spring, no sound effect

#### Additional context
This change only applies to `sound_effect` and `sound_effect_preload`. Seasonal playlists could also be added as a future PR.
